test/unit/pool/test-destroy-connection.js

```diff
@@ -11,10 +11,10 @@ server.listen(common.fakeServerPort, function (err) {
   pool.getConnection(function (err, connection) {
     assert.ifError(err);
 
-    assert.strictEqual(connection, pool._allConnections[0]);
+    assert.strictEqual(connection, pool._connectionManager._allConnections._map[1]);
```

Comment:
@dougwilson I really know how difficult it's to review many changes. I'm going to think about how I can break it up but, it's not easy. That's because `PoolConnectionManager` has all new functions. I wonder if it's possible to split this into multiple PRs easily.